shielddb-core/src/main/java/nf/fr/k49/shielddb/core/ShieldDBBuilder.java

```diff
@@ -21,14 +22,22 @@ public ShieldDBBuilder() {
 	public List<T> build() {
 		final BottomShield<T> bottom = new BottomShield<>(mapper, storage);
 		shields.add(bottom);
-		for(int i = 0; i < shields.size()-1 ; i++) {
-			final ShieldDBShield<T> cur = shields.get(i);
-			final ShieldDBShield<T> next = shields.get(i+1);
-			cur.setNextShield(next);
-		}
+		doNestShields();
 		return shields.get(0);
 	}
 
+	private void doNestShields() {
+		Iterator<ShieldDBShield<T>> shieldIterator = shields.iterator();
+
+		ShieldDBShield<T> bindTo = shieldIterator.next();
```

Comment:
Do you think "previousShield" could be a better name for the bindTo variable?

Reply:
should be. I similar topics I like using "outer" and "inner".
I'll change it